chromium_src/third_party/boringssl/src/crypto/curve25519/curve25519.c

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "third_party/boringssl/src/include/openssl/curve25519.h"
+
+#include "src/third_party/boringssl/src/crypto/curve25519/curve25519.c"
+
+#ifdef UNSAFE_BUFFERS_BUILD
+#pragma allow_unsafe_buffers
+#endif
+
+// https://datatracker.ietf.org/doc/html/rfc8032#section-5.1.5
+// requires scalar to follow this requirements 'The lowest 3 bits of the first
+// octet are cleared, the highest bit of the last octet is cleared, and the
+// second highest bit of the last octet is set'.
+int ED25519_is_scalar_pruned(const uint8_t scalar[32]) {
+  return (scalar[0] & 0b00000111) == 0b00000000 &&
+         (scalar[31] & 0b11000000) == 0b01000000;
+}
```

Comment:
If possible to change this file from `.c` to `.cc` please spanify.

Reply:
would that work having that `base` already depends on `boringssl`? 
https://source.chromium.org/chromium/chromium/src/+/main:base/BUILD.gn;l=1510;bpv=1

Reply:
Explained above, but yeah, it seems to be an obstacle. We could use `std::span`, but not with the file still being a `.c` in master.